### ISSUE TITLE
`Y066` cases that can't be rewritten

### ISSUE THREAD

**Issue by tusharsadhwani** [PyCQA/flake8-pyi#491]
Consider this case:

```python
import sys

if sys.version_info < (3, 8):  # Y066 use "if sys.version_info >= (3, 8)"
    def bar(x): ...
elif sys.version_info >= (3, 10):
    def bar(x, *, bar=True): ...
else:
    def bar(x, *, bar=True, baz=False): ...
```

I don't think it is possible to rewrite this code with `if sys.version_info >= (3, 8)`. Should Y066  be raised in this case?

**Comment by tusharsadhwani** [comment]
Specifically, `Y066`'s suggestions are only applicable if `Y066` is raised on every `if` and `elif` condition in the chain.

**Comment by AlexWaygood** [comment]
Can't your example be rewritten to

```py
if sys.version_info >= (3, 10):
    def bar(x, *, bar=True): ...
elif sys.version_info >= (3, 8):
    def bar(x, *, bar=True, baz=False): ...
else:
    def bar(x): ...
```

?

**Comment by tusharsadhwani** [comment]
My bad! You're right, the `else` clause is equivalent to `>= 3.8`.

**Comment by tusharsadhwani** [comment]
It does get a bit dicey in some cases, like:

```python
if sys.version_info < (3, 8):  # Y066 use "if sys.version_info >= (3, 8)"
    def bar(x): ...
elif sys.version_info == (3, 10):
    def bar(x, *, bar=True): ...
else:
    def bar(x, *, bar=True, baz=False): ...
```

is not equivalent to:

```python
if sys.version_info == (3, 10):
    def bar(x, *, bar=True): ...
elif sys.version_info >= (3, 8):
    def bar(x, *, bar=True, baz=False): ...
else:
    def bar(x): ...
```

For Python 3.11, it will execute the wrong block, but it's hard to argue about that one.

**Comment by AlexWaygood** [comment]
I think we have another rule that warns against `==` comparisons with `sys.version_info`, because `sys.version_info` is always a tuple longer than 2 elements, so `sys.version_info == (3, 10)` will always resolve to `False`
